### PR TITLE
e2e: add E2E test for block mode PVC cloning

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: examples/rbd/pvc-raw-block-clone.yaml
Comment (nixpanic) [inline examples/rbd/pvc-raw-block-clone.yaml:8]: this should be squashed into the commit that adds this file

**Review by nixpanic — CHANGES_REQUESTED** [review]

File: e2e/rbd.go
Comment (Madhu-1) [inline e2e/rbd.go:595]: is there any difference between this block of code and filesystem clone? If no I would like to move the code to the helper function and make use of it to avoid much larger code duplication

Comment (humblec) [reply e2e/rbd.go:595]: @Madhu-1 looking into the E2E , I see a good room for improvement in various things. We have to tackle it one by one. I was thinking to do that clean up or re factoring any way. I could start here or could start altogether in a new series. Please let me know the choice

File: examples/rbd/pvc-raw-block-clone.yaml
Comment (humblec) [reply examples/rbd/pvc-raw-block-clone.yaml:8]: yeah. sure.

File: e2e/rbd.go
Comment (humblec) [reply e2e/rbd.go:242]: our PVC file got Raw string in it. so kept in parity. Are we in agreement to change everything to have a common schema ? @nixpanic @Madhu-1

File: e2e/rbd.go
Comment (Madhu-1) [reply e2e/rbd.go:595]: Yes, there are many things to improve in E2E. I think this would be a good start.

File: e2e/rbd.go
Comment (Madhu-1) [reply e2e/rbd.go:242]: Yes instead of raw-block block good fine.

File: e2e/rbd.go
Comment (humblec) [reply e2e/rbd.go:531]: Yep, I will be doing code improvement in general for the repo.. Let us start here.  Introduced a helper function.

**Comment by humblec** [comment]
@Madhu-1 @nixpanic ptal .

**Comment by nixpanic** [comment]
> @Madhu-1 @nixpanic ptal .

**REMINDER:** Please click the round arrows in the upper-right box next to my name when re-requesting reviews. That way GitHub can list it in standard searches like ([Awaiting review from you](https://github.com/ceph/ceph-csi/pulls/review-requested/@me)).

File: examples/rbd/pvc-block-clone.yaml
Comment (nixpanic) [inline examples/rbd/pvc-block-clone.yaml:10]: please just use `block-pvc` like `volumeMode: Block` (it does not have `raw` in the value)

File: examples/rbd/block-pod-clone.yaml
Comment (nixpanic) [inline examples/rbd/block-pod-clone.yaml:5]: same here, no need to include `raw` in the name

File: e2e/rbd.go
Comment (nixpanic) [inline e2e/rbd.go:538]: I must have missed this earlier, but `rawPvcPath` is badly named too. No need to update it with this PR, unless you want/need to send a new version.

**Review by nixpanic — APPROVED** [review]
These commits belong tightly together and could be squashed into one:
- e2e: add block mode PVC clone file 
-  e2e: add pod yaml which consume block mode clone pvc

Looks good to me, not required to address the nits I mentioned (I leave that up to you)

File: examples/rbd/pvc-block-clone.yaml
Comment (humblec) [reply examples/rbd/pvc-block-clone.yaml:10]: yeah.. I will remove the  entire `raw` strings in next PR.

File: examples/rbd/block-pod-clone.yaml
Comment (humblec) [reply examples/rbd/block-pod-clone.yaml:5]: same as above, will take care of it in next PR.

File: e2e/rbd.go
Comment (humblec) [reply e2e/rbd.go:538]: Yeah, will put everything in next PR.. thanks

**Comment by humblec** [comment]
@nixpanic Thanks.. Will take care of the `nits` in next PR.. :+1: 
@Madhu-1 ptal.

File: e2e/utils.go
Comment (Madhu-1) [inline e2e/utils.go:503]: Does this function needs to be exported?

**Review by nixpanic — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**
